File: 2938-separate-black-and-white-balls/2938-separate-black-and-white-balls.cpp
class Solution {
public:
    long long minimumSteps(string s) {
        long long ans = 0;
        int blackBallCount = 0;
        
        // Iterate through each ball in the string
        for (const char& c : s){
            if (c == '0') ans += blackBallCount;
            else ++blackBallCount;
        }
        
        return ans;
    }
};

File: 2938-separate-black-and-white-balls/2938-separate-black-and-white-balls.py
class Solution:
    def minimumSteps(self, s: str) -> int:
        ans, black = 0, 0
        
        # Iterate through each ball in the string
        for c in s:
            if c == '0':
                ans += black
            else:
                black += 1
                
        return ans